ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -1269,13 +1262,25 @@ private void handleGetResponse(GetResponse getResponse, List<RemoteReplicaInfo>
         replicatingFromRemoteColo, replicatingOverSsl, datacenterName, remoteColoGetRequestForStandby);
   }
 
-  /**
-   * Applies a TTL update to the blob described by {@code messageInfo}.
-   * @param messageInfo the {@link MessageInfo} that will be transformed into a TTL update
-   * @param remoteReplicaInfo The remote replica that is being replicated from
-   * @throws StoreException
-   */
-  private void applyTtlUpdate(MessageInfo messageInfo, RemoteReplicaInfo remoteReplicaInfo) throws StoreException {
+  protected void applyPut(MessageSievingInputStream validMessageDetectionInputStream, RemoteReplicaInfo remoteReplicaInfo)
+      throws StoreException, IOException {
```

Comment:
The interface will be consistent with other apply* methods if we pass the `messageInfoList` in this method instead of creating `messageInfoList` inside the method.

Reply:
> Overall some javadocs are missing and there is no description in the PR explaining what is this about and why is this needed.
> 
> Also, what happens when a blob is replicated from on-prem but it is not present in recovered data from the backup because the recovery logic hasn't caught up enough with the backup yet? Or do you start `Backupcheckerthread` only after you are caught up with the backup? How would you decide that?

Added javadocs. And we discussed the idea behind this PR last Friday. I assumed you'd know. I will add it before final merge.
As for the second question, the code would throw an error and i'd log it in the logfile for now.

Reply:
I have explored the option of passing messageInfoList but that's not possible since MessageFormatWriteSet takes validMessageDetectionInputStream